ingestion/src/metadata/orm_profiler/sink/file.py

```diff
@@ -0,0 +1,100 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Profiler File Sink
+"""
+from pathlib import Path
+
+from metadata.config.common import ConfigModel
+from metadata.ingestion.api.common import Entity, WorkflowContext
+from metadata.ingestion.api.sink import Sink, SinkStatus
+from metadata.orm_profiler.api.models import ProfileAndTests
+from metadata.orm_profiler.utils import logger
+
+logger = logger()
+
+
+class FileSinkConfig(ConfigModel):
+    filename: str
+
+
+class FileSink(Sink[Entity]):
+    config: FileSinkConfig
+    report: SinkStatus
+
+    def __init__(
+        self,
+        ctx: WorkflowContext,
+        config: FileSinkConfig,
+    ):
+        super().__init__(ctx)
+        self.config = config
+        self.report = SinkStatus()
+
+        fpath = Path(self.config.filename)
+
+        # Build the path if it does not exist
+        if not fpath.parent.is_dir():
+            Path(self.config.filename).mkdir(parents=True, exist_ok=True)
+        self.file = fpath.open("w")
+        self.wrote_something = False
+
+    @classmethod
+    def create(cls, config_dict: dict, _, ctx: WorkflowContext):
+        config = FileSinkConfig.parse_obj(config_dict)
+        return cls(ctx, config)
+
+    def write_record(self, record: ProfileAndTests) -> None:
```

Comment:
jinja template could have helped you here :)

Reply:
yeah.. no the greatest solution, just needed something quick for the demo 😅

Reply:
I'll get this fixed after we apply the new test result classes 🙏